cpp/order/Steinhardt.cc

```diff
@@ -184,208 +185,196 @@ void Steinhardt::baseCompute(const box::Box& box,
+                            const vec3<float>* points)
+{
+    const size_t* neighbor_list(nlist->getNeighbors());
 
     const float rminsq = m_rmin * m_rmin;
     const float rmaxsq = m_rmax * m_rmax;
-    const float normalizationfactor = 4*M_PI/(2*m_l+1);
-
-    parallel_for(tbb::blocked_range<size_t>(0, m_Np),
-        [=] (const blocked_range<size_t>& r)
-		{
-		bool QlmAve_exists;
-		m_Qlm_local.local(QlmAve_exists);
-		if (!QlmAve_exists)
-			{
-            m_Qlm_local.local() = new complex<float> [2*m_l+1];
-            memset((void*) m_Qlm_local.local(), 0, sizeof(complex<float>) * (2*m_l+1));
-            }
+    const float normalizationfactor = 4 * M_PI / (2 * m_l + 1);
 
-		size_t bond(nlist->find_first_index(r.begin()));
-		// for each reference point
-		for (unsigned int i = r.begin(); i != r.end(); i++)
-			{
-			const vec3<float> ri = points[i];
-			unsigned int neighborcount(1);
-
-			for (; bond < nlist->getNumBonds() && neighbor_list[2*bond] == i; ++bond)
-				{
-				const unsigned int n(neighbor_list[2*bond + 1]);
-
-				if (n == i)
-					{
-					continue;
-					}
-
-				const vec3<float> rn = points[n];
-				// rin = rn - ri, from i pointing to j.
-				const vec3<float> rin = box.wrap(rn - ri);
-				const float rinsq = dot(rin, rin);
-
-				if (rinsq < rmaxsq && rinsq > rminsq)
-					{
-					size_t neighborhood_bond(nlist->find_first_index(n));
-					for (; neighborhood_bond < nlist->getNumBonds() && neighbor_list[2*neighborhood_bond] == n; ++neighborhood_bond)
-						{
-						const unsigned int j(neighbor_list[2*neighborhood_bond + 1]);
-
-						if (n == j)
-							{
-							continue;
-							}
-
-						// rnj = rj - rn, from n pointing to j.
-						const vec3<float> rnj = box.wrap(points[j] - rn);
-						const float rnjsq = dot(rnj, rnj);
-
-						if (rnjsq < rmaxsq && rnjsq > rminsq)
-							{
-							for (unsigned int k = 0; k < (2*m_l+1); ++k)
-								{
-								// Adding all the Qlm of the neighbors
-								m_QlmiAve.get()[(2*m_l+1)*i+k] += m_Qlmi.get()[(2*m_l+1)*j+k];
-								}
-							neighborcount++;
-							}
-						} // End loop over particle neighbor's bonds
-					}
-				} // End loop over particle's bonds
-
-			// Normalize!
-			for (unsigned int k = 0; k < (2*m_l+1); ++k)
-				{
-				const unsigned int index = (2*m_l+1) * i + k;
-				// Adding the Qlm of the particle i itself
-				m_QlmiAve.get()[index] += m_Qlmi.get()[index];
-				m_QlmiAve.get()[index] /= neighborcount;
-				m_Qlm_local.local()[k] += m_QlmiAve.get()[index] / float(m_Np);
-				// Add the norm, which is the complex squared magnitude
-				m_QliAve.get()[i] += norm(m_Qlmi.get()[index]);
-				}
-			m_QliAve.get()[i] *= normalizationfactor;
-			m_QliAve.get()[i] = sqrt(m_QliAve.get()[i]);
-			} // Ends loop over particles i for Qlmi calcs
-		}); //End parallel function
-    }
+    parallel_for(tbb::blocked_range<size_t>(0, m_Np), [=](const blocked_range<size_t>& r) {
+        bool QlmAve_exists;
```

Comment:
Suggesting a change from `QlmAve_exists` to `Qlm_exists` as above, so it's clear these represent the same thing.